msal4j-sdk/src/test/java/com/microsoft/aad/msal4j/AcquireTokenSilentlyTest.java

```diff
@@ -116,6 +119,79 @@ void confidentialAppAcquireTokenSilently_claimsSkipCache() throws Throwable {
         assertInstanceOf(MsalInteractionRequiredException.class, ex.getCause());
     }
 
+    @Test
+    void testTokenRefreshReasons() throws Exception {
+        DefaultHttpClient httpClientMock = mock(DefaultHttpClient.class);
+
+        ConfidentialClientApplication cca =
+                ConfidentialClientApplication.builder("clientId", ClientCredentialFactory.createFromSecret("password"))
+                        .authority("https://login.microsoftonline.com/tenant/")
+                        .instanceDiscovery(false)
+                        .validateAuthority(false)
+                        .httpClient(httpClientMock)
+                        .build();
+
+        HashMap<String, String> responseParameters = new HashMap<>();
+
+        //Acquire a token that expired an hour ago
+        responseParameters.put("access_token", "expiredToken");
+        responseParameters.put("id_token", TestHelper.createIdToken(new HashMap<>()));
```

Comment:
Client Credentails doesn't respond with id token.

Reply:
Good point, it's valid as far the library is concerned but probably not a good practice. In the latest commit this test uses the OBO flow, which is more appropriate for a refresh test.